File: frontend/Assets/Scripts/DataTransfer/Server/NewTurnCmd.cs
using System;
using System.IO;
using Attributes;
using Utils;


namespace DataTransfer.Server {

    [DTO (DTOCode.NewTurn)]
    public class NewTurnCmd : ServerCommand {

        public static event Action <NewTurnCmd> OnReceived;

        public int Player { get; private set; }


        public override void ReadMembers (BinaryReader reader) {
            Player = reader.ReadInt32 ();
        }


        public override void Execute () {
            OnReceived._ (this);
        }

    }

}

File: backend/src/main/java/dto/server/NewTurnCmd.java
package dto.server;

import dto.DTOCode;
import dto.DTOs;
import io.netty.buffer.ByteBuf;


@DTOCode(DTOs.NEW_TURN)
public class NewTurnCmd extends ServerCommand {

    private final int playerId;


    public NewTurnCmd (int playerId) {
        this.playerId = playerId;
    }


    @Override
    public void writeMembers (ByteBuf byteBuf) {
        byteBuf.writeInt(playerId);
    }
}
